### PR TITLE
Adding functions to handle custom agent pools.

### LINKED ISSUE #53: Function: Get/New-ADOPSElasticPool
We need to create new functions for life-cycle management of Elastic Pools in Azure DevOps.

This issue will cover the get/create/new functions.

### LINKED ISSUE #55: Get-ADOPSPool
New Function to get/list Azure DevOps Agent pools.

### LINKED ISSUE #57: Function: Get-ADOPSNode
We need a new function to get/list Azure DevOps node information.
Docs: https://docs.microsoft.com/en-us/rest/api/azure/devops/distributedtask/nodes/list?view=azure-devops-rest-7.1

### PR DESCRIPTION
**Pull request by SebastianClaesson**
# Overview/Summary

Adding new functions to allow functionality to manage custom agent pools.
This allows you to pin an agent pool to a Azure Virtual Machine Scale Set.

I've added some examples of how to run the New-AZDOPSElasticPool, however I'll need to dig into PlayPS to create this using the same tooling used across the project.

No tests have been created as of yet.

## This PR fixes/adds/changes/removes

Adding new functions to:
- Create a new ElasticPool
- Get/List ElasticPool
- Get/List Node
- Get/List Pool

Closes #53 
Closes #57
Closes #55

## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/bjompen/AZDOPS/pulls)
- [X] Associated it with relevant [issues](https://github.com/bjompen/AZDOPS/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/bjompen/AZDOPS/tree/main)
- [] Performed testing and provided evidence.
- [Half X!] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by SebastianClaesson** [comment]
Resolving the conflig in azdops.psd1.
Added the functions that has been commited to main and not included when these functions were written.

**Comment by bjompen** [comment]
Please add basic tests to the functions as well as documentation,
Or just documentation so its easier for me to add tests afterwards.

File: Source/Public/Get-AZDOPSElasticPool.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+function Get-AZDOPSElasticPool {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]$Organization,
```
Comment (bjompen) [inline Source/Public/Get-AZDOPSElasticPool.ps1:5]: If no organization is given the URL will not work. Please change this to mandatory, or use the $org['organization'] variable in the URL as that one will always be populated.

File: Source/Public/Get-AZDOPSNode.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+function Get-AZDOPSNode {
+    [CmdletBinding()]
+    param (
+        [Parameter()]
+        [string]$Organization,
```
Comment (bjompen) [inline Source/Public/Get-AZDOPSNode.ps1:5]: If no organization is given the URL will not work. Please change this to mandatory, or use the $org['organization'] variable in the URL as that one will always be populated.

File: Source/Public/Get-AZDOPSPool.ps1 [hunk 3]
```diff
@@ -0,0 +1,51 @@
+function Get-AZDOPSPool {
+    [CmdletBinding(DefaultParameterSetName = 'All')]
+    param (
+        [Parameter(ParameterSetName = 'PoolId')]
+        [Parameter(ParameterSetName = 'PoolName')]
+        [Parameter(ParameterSetName = 'All')]
+        [string]$Organization,
```
Comment (bjompen) [inline Source/Public/Get-AZDOPSPool.ps1:7]: If no organization is given the URL will not work. Please change this to mandatory, or use the $org['organization'] variable in the URL as that one will always be populated.

File: Source/Public/Get-AZDOPSNode.ps1 (hunk 2 above)
Comment (bjompen) [reply Source/Public/Get-AZDOPSNode.ps1:5]: Would probalby help if i read the code right though.. sorry.. :/

File: Source/Public/Get-AZDOPSPool.ps1 (hunk 3 above)
Comment (bjompen) [reply Source/Public/Get-AZDOPSPool.ps1:7]: Would probably help if i read the code right though.. sorry.. :/

**Comment by StefanIvemo** [comment]
@SebastianClaesson ping me and I can help you get started with PlatyPS. There are guidelines in the contribution guide that might help you get started.

**Comment by SebastianClaesson** [comment]
Closing pull request as a new one with correct documentation and tests will be opened from my fork instead.

**Pull request closed without merging**